src/hooks/routing/useRouterTrade.ts

```diff
@@ -29,32 +30,39 @@ export function useRouterTrade(
   amountSpecified: CurrencyAmount<Currency> | undefined,
   currencyIn: Currency | undefined,
   currencyOut: Currency | undefined,
-  routerPreference: RouterPreference,
-  routerUrl?: string
+  quoteConfig: QuoteConfig
 ): {
   state: TradeState
   trade?: InterfaceTrade
   gasUseEstimateUSD?: CurrencyAmount<Token>
 } {
   const { provider } = useWeb3React()
+  const routerUrl = useAtomValue(swapRouterUrlAtom)
   const queryArgs = useGetQuoteArgs(
-    { provider, tradeType, amountSpecified, currencyIn, currencyOut, routerPreference, routerUrl },
-    /*skip=*/ routerPreference === RouterPreference.SKIP
+    {
+      provider,
+      tradeType,
+      amountSpecified,
+      currencyIn,
+      currencyOut,
+      routerPreference: quoteConfig.preference,
```

Comment:
At this point, could you just pass `quoteConfig` as the second arg (instead of passing `routerPreference` or `skip`), to keep it more semantic?